src/components/CheckboxWithLabel.js

```diff
@@ -23,26 +27,46 @@ const propTypes = {
     /** Component to display for label */
     LabelComponent: PropTypes.func,
 
-    /** Should the input be styled for errors  */
-    hasError: PropTypes.bool,
 
     /** Error text to display */
     errorText: PropTypes.string,
+
+    /** Indicates that the input is being used with the Form component */
+    isFormInput: PropTypes.bool,
+
+    /**
+     * The ID used to uniquely identify the input
+     *
+     * @param {Object} props - props passed to the input
+     * @returns {Object} - returns an Error object if isFormInput is supplied but inputID is falsey or not a string
+     */
+    inputID: props => FormUtils.getInputIDPropTypes(props),
+
+    /** Saves a draft of the input value when used in a form */
+    shouldSaveDraft: PropTypes.bool,
 };
 
 const defaultProps = {
+    onChange: () => {},
+    isFormInput: false,
+    inputID: undefined,
     style: [],
     label: undefined,
     LabelComponent: undefined,
-    hasError: false,
     errorText: '',
+    shouldSaveDraft: false,
 };
 
-const CheckboxWithLabel = (props) => {
+const CheckboxWithLabel = React.forwardRef((props, ref) => {
     const LabelComponent = props.LabelComponent;
     const defaultStyles = [styles.flexRow, styles.alignItemsCenter];
     const wrapperStyles = _.isArray(props.style) ? [...defaultStyles, ...props.style] : [...defaultStyles, props.style];
 
+    function toggleCheckbox() {
```

Comment:
@thesahindia let's move it outside the component?

Reply:
Umm, It's using props shouldn't we keep it inside.

Reply:
We can pass the props as an argument

Reply:
Should I use it like this? `onPress={() => toggleCheckbox(props.onPress, props.onChange)}`
otherwise getting this eslint error `PropType is defined but prop is never used` for `onPress` and `onChange`

Reply:
Hmm, that would get messy. Let's leave it as it is